src/hooks/useFirestoreCollectionWithAtom.ts

```diff
@@ -191,6 +196,10 @@ export function useFirestoreCollectionWithAtom<T = TableRow>(
               available: docs.length >= memoizedQuery.limit,
             }));
           }
+          // on each new snapshot, use the query to get and set the document count from the server
+          getCountFromServer(memoizedQuery.query).then((value) => {
```

Comment:
the `memoizedQuery.query` already has a limit apply to it causing the getCountFromServer function only returns the number of total loaded documents, rather than total number documents in the collection. I'm not sure if you can remove the limit from `memoizedQuery.query` otherwise you can just recreate the query with collection path and filters only